platform/ios/MapboxGL.m

```diff
@@ -0,0 +1,48 @@
+#import <Foundation/Foundation.h>
+
+#import "MapboxGL.h"
+#import "NSProcessInfo+MGLAdditions.h"
+
+@interface MapboxGL()
+
+@property (atomic) NSString *accessToken;
+
+@end
+
+
+@implementation MapboxGL
+
+static MapboxGL *_sharedManager;
+
+// Can be called from any thread. Called implicitly from any
+// public class convenience methods.
+//
++ (id) sharedInstanceWithAccessToken:(NSString *)token {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        if ( ! NSProcessInfo.processInfo.mgl_isInterfaceBuilderDesignablesAgent) {
+            void (^setupBlock)() = ^{
+                _sharedManager = [[self alloc] init];
+                _sharedManager.accessToken = token;
+            };
+            if ( ! [[NSThread currentThread] isMainThread]) {
+                dispatch_sync(dispatch_get_main_queue(), ^{
+                    setupBlock();
+                });
+            }
+            else {
+                setupBlock();
+            }
+        }
+    });
+    return _sharedManager;
+}
+
++ (NSString *) getAccessToken {
```

Comment:
I ended up going with just renaming to `+sharedAccessToken` to avoid any `get` convention.  The instance methods route wouldn't work due to [MGLMapView needing to retrieve it](https://github.com/mapbox/mapbox-gl-native/pull/1329/files#diff-f315cd6ead3270a0818a2dcfb84bd99bR126).  By keeping it in a `+sharedManager` this class stays more extensible.